kernel/kernel-api/src/main/java/io/delta/kernel/internal/stats/FileSizeHistogram.java

```diff
@@ -54,6 +54,15 @@ public static FileSizeHistogram createDefaultHistogram() {
     return new FileSizeHistogram(defaultBoundaries, zeroCounts, zeroBytes);
   }
 
+  public static Optional<FileSizeHistogram> fromRow(Row row) {
+    requireNonNull(row);
+    if (!FULL_SCHEMA.equals(row.getSchema())) {
+      return Optional.empty();
+    }
```

Comment:
why do we need this check? if the schemas don't match, we should throw error right?

Reply:
+1